MultiLangProgGuide/MultiLangProgGuide.rst

```diff
@@ -1819,6 +1821,920 @@ If the requested :code:`timeout_nsec` exceeds
+  the method has a return code of :code:`C.YDB_OK` and
+  :code:`sub.GetLenUsed()` returns zero (i.e., the empty string).
+
+If there is an error, :code:`SubNextS()` has an `error return code`_
+prefixed with :code:`C.`.
+
+Go SubPrevS()
+-------------
+
+.. code-block:: go
+
+    SubPrevS(sub *BufferT) int
+
+:code:`SubPrevS()` wraps `ydb_subscript_previous_s()` to facilitate
+breadth-first traversal of a local or global variable sub-tree in
+reverse order.
+
+- If there is a node (either a node with data, or a subtree with or
+  without a node) in the subtree with a subscript preceding that of
+  the location specified by the :code:`KeyT` structure (neither a node
+  nor a subtree needs to exist at the location specified by the
+  :code:`KeyT` structure), :code:`SubNextS()` has a return code of
+  :code:`C.YDB_OK` and :code:`sub` references the value of that
+  subscript.
+- If the last subscript of the location specified by the :code:`KeyT`
+  structure is the empty string, :code:`sub` references the subscript
+  value of the last node of the subtree.
+- If there is no previous node at that level of the subtree, the method
+  has a return code of :code:`C.YDB_OK` and :code:`sub.GetLenUsed()`
+  returns zero (i.e., the empty string).
+
+Note that in the event an application uses empty string subscripts,
+and :code:`sub.GetLenUsed()` is zero, the application may need to
+invoke :code:`DataS()` to determine whether a node or subtree exists
+at that location.
+
+If there is an error, :code:`SubNextS()` has an `error return code`_
+prefixed with :code:`C.`.
+
+-----------------------
+Go Simple API Functions
+-----------------------
+
+Go LockS()
+----------
+
+.. code-block:: go
+
+    LockS( timeoutNsec uint64, ... *KeyT) int
+
+The function wraps `ydb_lock_s()`. It releases all `Locks`_ held by
+the process, and acquires the locks whose resource names are specified
+by the :code:`KeyT` parameters; if there are none, it simply releases
+all locks and returns immediately.
+
+The :code:`timeoutNsec` parameter can take a value from zero through
+:code:`C.YDB_MAX_TIME_NSEC`. If :code:`timeoutNsec` exceeds
+:code:`C.YDB_MAX_TIME_NSEC`, the function immediately returns with an
+error code of :code:`C.YDB_ERR_TIME2LONG`.  If within the specified
+time, :code:`LockS()` is able to acquire all requested locks, it does
+so, and returns with a return code of :code:`C.YDB_OK`; otherwise it
+acquires no locks and returns with a return code of
+:code:`C.YDB_LOCK_TIMEOUT`. If :code:`timeoutNsec` is zero, the
+function makes exactly one attempt to acquire the locks. Errors result
+an `error return code`_ prefixed with :code:`C.`.
+
+Go Comprehensive API
+====================
+
+The Go Comprehensive API is a project for the future, to follow the C
+`Comprehensive API`_
+
+Go Utility Functions and Methods
+================================
+
+-----------------------
+Go KeyT Utility Methods
+-----------------------
+
+GetSubBufAlloc(n)
+GetSubBufUsed(n)
```

Comment:
Not understanding why these are "Utility Methods" instead of structure methods.

Reply:
You read past where I had material to review. This is an area where I made notes to myself. The next round of reviews will have something here worth reviewing.